openhcl/minimal_rt/src/rt.rs

```diff
@@ -23,8 +23,10 @@ mod instead_of_builtins {
         fn memcpy(dest: *mut u8, src: *const u8, n: usize) -> *mut u8;
     }
 
-    /// Implementation cribbed from compiler_builtins.
+    // SAFETY: The minimal_rt_build crate ensures that when this code is compiled
+    // there is no libc for this to conflict with.
     #[unsafe(no_mangle)]
+    /// Implementation cribbed from compiler_builtins.
```

Comment:
lets ensure that doc comments live above attributes, in keeping with convention

Reply:
That looked worse to me. Since the safety comments need to be immediately above the attribute, you get:

```
/// doc comment
// safety comment
#[attr]
unsafe fn foo
```

I didn't really like how the two comments were mixed right next to each other.

Reply:
That looks fine to me, and its something I've seen done (and done myself) elsewhere in the codebase.